usvm-ts-dataflow/src/main/kotlin/org/usvm/dataflow/ts/ifds/EtsForwardIfdsRunner.kt

```diff
@@ -0,0 +1,76 @@
+package org.usvm.dataflow.ts.ifds
+
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.isActive
+import org.jacodb.ets.model.EtsStmt
+import org.jacodb.ets.model.EtsMethod
+import org.usvm.dataflow.ifds.Analyzer
+import org.usvm.dataflow.ifds.Edge
+import org.usvm.dataflow.ifds.IfdsResult
+import org.usvm.dataflow.ifds.QueueEmptinessChanged
+import org.usvm.dataflow.ifds.Reason
+import org.usvm.dataflow.ifds.Runner
+import org.usvm.dataflow.ifds.SingletonUnit
+import org.usvm.dataflow.ifds.UnitType
+import org.usvm.dataflow.ts.graph.EtsApplicationGraph
+import org.usvm.dataflow.ts.infer.AnalyzerEvent
+import org.usvm.dataflow.ts.infer.TypeInferenceManager
+import org.usvm.dataflow.ts.util.EtsTraits
+
+class EtsForwardIfdsRunner<Fact, Event : AnalyzerEvent>(
+    override val graph: EtsApplicationGraph,
+    val analyzer: Analyzer<Fact, Event, EtsMethod, EtsStmt>,
+    val traits: EtsTraits,
+    val manager: TypeInferenceManager,
+) : Runner<Fact, EtsMethod, EtsStmt> {
+    val queue: ArrayDeque<EtsForwardMethodRunner<Fact, Event>> = ArrayDeque()
+
+    private val queueIsEmpty = QueueEmptinessChanged(runner = this, isEmpty = true)
+
+    private val methods = graph.cp.projectAndSdkClasses.flatMap { it.methods + it.ctor }
```

Comment:
Depending on the AA used, `::methods` already contains a constructor (if it was present in the IR).